### PR TITLE
Always ignore ``GenericRelation`` in page copy

### PR DESCRIPTION
**Pull request by jsma**
In https://github.com/wagtail/wagtail/issues/3920#issuecomment-479428695 @kaedroho said:
> Just hit this issue myself, the error message given doesn't make it clear what's going on. So I wonder if it's worth excluding generic relations by default?

This PR aims to do just that. 

I was bitten by this issue recently and a related issue prevented our ``exclude_fields_in_copy`` from being respected when editors copy pages in the Wagtail admin (a fix for that available in #8048).

### CODE REVIEW

File: wagtail/tests/test_page_model.py [hunk 1]
```diff
@@ -1969,6 +1971,23 @@ def test_copy_page_with_additional_excluded_fields(self):
...
+            instance=PageWithGenericRelation(
+                title="PageWithGenericRelation",
+                slug="page-with-generic-relation",
+                live=True,
+                has_unpublished_changes=False,
+            )
+        )
+        original_page.copy(
+            to=homepage, update_attrs={"slug": f"{original_page.slug}-2"}
+        )
```
[the remark this answers is not in the document]
Comment (jsma) [reply wagtail/tests/test_page_model.py:1989]: Ah, this was the original test to prove the bug existed but I guess I forgot to revisit it to make it more than just a test that would detect future regressions. Will push an update shortly.

[the remark this answers is not in the document]
Comment (jsma) [reply wagtail/tests/test_page_model.py:1992]: I've updated the test to actually create a related item and then assert that it is not copied to the new page copy. Let me know if you want me to squash these commits. I had intended to just commit the failing test first and then solve the problem in a single follow-up commit and we see how that went...

Thanks for catching this and sorry for forgetting to update the test previously!

**Review by zerolab — APPROVED** [review]
LGTM.

Thank you @jsma

**Comment by zerolab** [comment]
Thank you @jsma, Merged in e7758d58cb8de57a78cf55c23f668d9b5f059533

**Pull request closed without merging**